src/components/backup/RestoreCloudStep.tsx

```diff
@@ -186,16 +217,26 @@ export default function RestoreCloudStep() {
           const firstAddress = firstWallet ? (firstWallet.addresses || [])[0].address : undefined;
           const p1 = dispatch(walletsSetSelected(firstWallet));
           const p2 = dispatch(addressSetSelected(firstAddress));
-          await Promise.all([p1, p2]);
-          await initializeWallet(null, null, null, false, false, null, true, null);
+          const p3 = dispatch(fetchWalletNames());
+          const p4 = profilesEnabled ? dispatch(fetchWalletENSAvatars()) : null;
```

Comment:
I think these two lines were removed by me as part of a perf improvement and are being added back due to a merge conflict that wasn't resolved correctly